compiler/rustc_mir_transform/src/coverage/spans.rs

```diff
@@ -466,11 +468,16 @@ impl<'a> CoverageSpansGenerator<'a> {
                 previous iteration, or prev started a new disjoint span"
             );
             if dup.span.hi() <= self.curr().span.lo() {
-                let pending_dups = self.pending_dups.split_off(0);
-                for dup in pending_dups.into_iter() {
+                // Temporarily steal `pending_dups` into a local, so that we can
+                // drain it while calling other self methods.
+                let mut pending_dups = std::mem::take(&mut self.pending_dups);
+                for dup in pending_dups.drain(..) {
                     debug!("    ...adding at least one pending={:?}", dup);
                     self.push_refined_span(dup);
                 }
+                // The list of dups is now empty, but we can recycle its capacity.
+                self.pending_dups = pending_dups;
+                assert!(self.pending_dups.is_empty());
```

Comment:
Should we also assert before the assignment, to verify that nobody added to the field while we were not looking?